File: commerce-order-engine-java/src/main/java/com/example/commerce/order/inventory/InventoryModels.java
package com.example.commerce.order.inventory;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import java.time.Instant;
import java.util.List;

public final class InventoryModels {
    private InventoryModels() {}
    public record Inventory(long productId, long availableQuantity, long reservedQuantity, long version, Instant createdAt, Instant updatedAt) {}
    public record Movement(long id, long productId, String movementType, long quantityDelta, long availableAfter, String reason, Instant createdAt) {}
    public record InventoryPage(List<Inventory> items, int page, int pageSize, long totalCount) {}
    public record CreateInventoryRequest(@NotNull Long productId, @Min(0) long initialQuantity) {}
    public record AdjustInventoryRequest(long quantityDelta, @NotBlank @Size(max=200) String reason) {}
}

File: commerce-operations-dotnet/src/Commerce.Operations.Api/Inventories/InventoryModels.cs
using Commerce.Operations.Api.Products;

namespace Commerce.Operations.Api.Inventories;

public sealed record InventoryRecord(long ProductId, long AvailableQuantity, long ReservedQuantity, long Version, DateTime CreatedAt, DateTime UpdatedAt);
public sealed record InventoryPage(IReadOnlyList<InventoryRecord> Items, int Page, int PageSize, long TotalCount);
public sealed record InventoryView(long ProductId, string? Sku, string? ProductName, long AvailableQuantity, long ReservedQuantity, long Version, DateTime UpdatedAt);
public sealed record InventoryViewPage(IReadOnlyList<InventoryView> Items, int Page, int PageSize, long TotalCount);
public sealed record InventoryMovement(long Id, long ProductId, string MovementType, long QuantityDelta, long AvailableAfter, string Reason, DateTime CreatedAt);
public sealed record CreateInventoryRequest(long ProductId, long InitialQuantity);
public sealed record AdjustInventoryRequest(long QuantityDelta, string Reason);

